msm/src/serialization/witness.rs

```diff
@@ -112,6 +140,44 @@ impl<Fp: PrimeField> Env<Fp> {
             }
         }
     }
+
+    pub fn add_rangecheck4_table_value(&mut self, i: usize) {
+        if i < (1 << 4) {
+            self.lookup_t_multiplicities_rangecheck4[i] += Fp::one();
+        } else {
+            self.lookup_t_multiplicities_rangecheck4[0] += Fp::one();
```

Comment:
Is this dummy value handling? Wondering if it's not bringing any soundness issues, since in the first `if` case you also add to the `[0]` element, right?

Reply:
See equation 14: https://eprint.iacr.org/2022/1530.pdf